smarts/sstudio/genhistories.py

```diff
@@ -229,8 +233,16 @@ def __init__(self, dataset_spec, output):
             "position_y": "x" if self._swap_xy else "y",
         }
 
+    def check_dataset_spec(self, dataset_spec: Dict[str, Any]):
+        super().check_dataset_spec(dataset_spec)
+        hiw = dataset_spec.get("heading_inference_window", 2)
+        if hiw != 2:
+            raise ValueError(
+                "heading_inference_window not yet supported for Interaction datasets."
+            )
```

Comment:
Is this to reserve this argument?

Reply:
The reason I put this there is because headings are inferred for both the NGSIM and Interaction, so one might assume that the 3 heading-inference-related YAML options that I added here would work for both of these. 

However, since (a) it's unclear if the nature of the noise in Interaction is the same as in NGSIM and (b) this particular option (`heading_inference_window`) would require non-trivial changes to the way the `Interaction` subclass works, I instead added this exception so that if someone tries to add `heading_inference_window` to their yaml file for an Interaction dataset (like they have in, say, their NGSIM yaml files) they can be made aware that "yes, that option _could_ be added here in the future if you think you need it, but we haven't done it yet."

The reason 2 is the default value there is because that's basically the "no window" case.

Reply:
Just added a comment about this.